homeassistant/components/dyson/fan.py

```diff
@@ -206,12 +203,28 @@ def device_state_attributes(self) -> dict:
             ATTR_DYSON_SPEED_LIST: self.dyson_speed_list,
         }
 
-    def set_speed(self, speed: str) -> None:
-        """Set the speed of the fan."""
-        if speed not in SPEED_LIST_HA:
-            raise ValueError(f'"{speed}" is not a valid speed')
-        _LOGGER.debug("Set fan speed to: %s", speed)
-        self.set_dyson_speed(SPEED_HA_TO_DYSON[speed])
+    def set_auto_mode(self, auto_mode: bool) -> None:
+        """Set auto mode."""
+        raise NotImplementedError
+
+    def set_percentage(self, percentage: int) -> None:
+        """Set the speed percentage of the fan."""
+        if percentage is None:
```

Comment:
We should not accept `None` as percentage as it should be an integer. The caller should make sure to pass an integer.

Reply:
fixed